File: S1-voluntario/FactoriaCarretera.rb
require('./S1/FactoriaCarreraYBicicleta.rb');

module S1
	public
	class FactoriaCarretera < FactoriaCarreraYBicicleta

		public
		def crearBicicleta(valor)			
			BiciCarretera.clone(valor)
		end

		public
		def crearCarrera
			CarreraCarretera.clone
		end
	end
end

File: P1/S1/FactoriaCarretera.java
package S1;

public class FactoriaCarretera extends FactoriaCarreraYBicicleta {

	@Override
	public Bicicleta crearBicicleta() {
		return new BiciCarretera();
	}

	@Override
	public Carrera crearCarrera() {
		return new CarreraCarretera();
	}

}